tools/jit/templates/aten_dispatch.cpp

```diff
@@ -142,6 +142,113 @@ std::array<bool, N> as_bool_array(const std::vector<int64_t>& vec) {
   return res;
 }
 
+template<typename T>
+T tensor_as(const at::Tensor& t) {
+  throw std::runtime_error("Invalid tensor_as overload");
+}
+
+template<>
+int64_t tensor_as(const at::Tensor& t) {
+  if (t.numel() > 1)
+    throw std::runtime_error("Trying to convert a tensor with more than 1 element into a scalar");
+  // TODO: transfer to CPU
+  switch (t.type().scalarType()) {
```

Comment:
We can replace these with at::Scalar(t), the control flow PR shows how to do it.